flink-python/pyflink/table/functions.py

```diff
@@ -138,6 +222,528 @@ def merge(self, accumulator, accumulators):
+                    order_list = []
+                order_list.append(order)
+                value_to_order_map[value] = order_list
+            else:
+                order = args[1]
+
+            if prev_order is None or prev_order <= order:
+                accumulator[0] = value
+                accumulator[1] = order
+
+            if order in order_to_value_map:
+                value_list = order_to_value_map[order]
+            else:
+                value_list = []
+            value_list.append(value)
+            order_to_value_map[order] = value_list
+
+    def retract(self, accumulator, *args):
+        if args[0] is not None:
+            value = args[0]
+            prev_value = accumulator[0]
+            prev_order = accumulator[1]
+            value_to_order_map = accumulator[2]
+            order_to_value_map = accumulator[3]
+
+            # get the order of current value if not given
+            if len(args) == 1:
+                if value in value_to_order_map and len(value_to_order_map[value]) > 0:
+                    order_list = value_to_order_map[value]
+                else:
+                    # this data has not been accumulated
+                    return
+                # get and remove current order in value_to_order_map
+                order = order_list.pop(0)
+                if len(order_list) == 0:
+                    del value_to_order_map[value]
+                else:
+                    value_to_order_map[value] = order_list
+            else:
+                order = args[1]
+
+            if order in order_to_value_map:
+                value_list = order_to_value_map[order]
+            else:
+                return
+
+            if value in value_list:
+                value_list.remove(value)
+                if len(value_list) == 0:
+                    del order_to_value_map[order]
+                else:
+                    order_to_value_map[order] = value_list
+
+            if value == prev_value:
+                start_key = prev_order
+                next_key = MIN_LONG_VALUE
+                for key in order_to_value_map:
+                    if start_key >= key > next_key:
+                        next_key = key
+
+                if next_key != MIN_LONG_VALUE:
+                    values = order_to_value_map[next_key]
+                    accumulator[0] = values[len(values) - 1]
+                    accumulator[1] = next_key
+                else:
+                    accumulator[0] = None
+                    accumulator[1] = None
+
+    def merge(self, accumulator, accumulators):
+        raise NotImplementedError("This function does not support merge.")
+
+
+class ListAggFunction(AggregateFunction):
+
+    def get_value(self, accumulator):
+        if accumulator[1] is None:
+            return None
+        else:
+            return accumulator[0].join(accumulator[1])
+
+    def create_accumulator(self):
+        # delimiter, values
+        return [',', None]
```

Comment:
What about changing the initial value from `[',', None]` to `[',', []]`? It can help avoid the logic of deciding whether accumulator[1] is None